bgbb_airflow/pred_airflow_job.py

```diff
@@ -1,50 +1,70 @@
+import datetime as dt
 from os.path import join
-from typing import List, Tuple, Union
+from typing import Dict, List, Tuple, Union
 
 import click
+import pandas as pd
 from bgbb import BGBB
-from bgbb.sql.bgbb_udfs import mk_n_returns_udf, mk_p_alive_udf
-from bgbb.sql.sql_utils import run_rec_freq_spk
+from bgbb.sql.bgbb_udfs import add_p_th, mk_n_returns_udf, mk_p_alive_udf
+from bgbb.sql.sql_utils import run_rec_freq_spk, S3_DAY_FMT_DASH
 from pyspark.sql import SparkSession
 
 from bgbb_airflow.bgbb_utils import PythonLiteralOption
 
+pd.options.display.max_columns = 20
+pd.options.display.width = 120
+Dash_str = str
+
 default_pred_bucket = "s3://net-mozaws-prod-us-west-2-pipeline-analysis"
-default_pred_prefix = "wbeard/bgbb_preds"
+default_pred_prefix = "wbeard/active_profiles"
 default_param_bucket = default_pred_bucket
 default_param_prefix = "wbeard/bgbb_params"
 
 
-def pull_most_recent_params(spark, param_bucket, param_prefix):
+def pull_most_recent_params(spark, max_sub_date, param_bucket, param_prefix):
     pars_loc = join(param_bucket, param_prefix)
     print("Reading params from {}".format(pars_loc))
     spars = spark.read.parquet(pars_loc)
-    pars_df = spars.orderBy(spars.submission_date_s3.desc()).limit(1).toPandas()
+    pars_df = (
+        spars.filter(spars.submission_date_s3 <= max_sub_date)
```

Comment:
`submission_day_s3` is `YYYY-MM-DD` or `YYYYMMDD`?

Reply:
`YYYY-MM-DD`. I annotated it as a `Dash_str` type [here below](https://github.com/wcbeard/bgbb_airflow/pull/2/files#diff-a6f1f032764b436f766c42b0d3a66971R40). Should I annotate in this arg as well, or is there a better way to specify this?

Reply:
I think the most human friendly way is to add the expected date format as a help string, but after re-reading, this looks fine.

Reply:
Ok, I added a help string in commit below